src/services/chat.service.js

```diff
@@ -0,0 +1,28 @@
+import { Op } from 'sequelize';
+import db from '../database/models/index';
+
+const ChatService = {
+  async addChat(chat) {
+    const newChat = await db.Chats.create(chat);
+    return newChat;
+  },
+
+  async getChat(userId, offset, limit) {
+    const chats = await db.Chats.findAll({
+      where: {
+        [Op.or]: [
+          { authorId: userId },
+          { receiverId: null }
+        ]
+      },
+      offset,
+      limit,
+      order: [
+        ['id', 'DESC']
```

Comment:
if the user provides something like:
```
{
 offset: 102,
 limit: 10,
}
```
it will still return all the chats, even tho that id(`offset`) does not exist
or
```
{
 offset: 1,
 limit: 2,
}
``` 
is still returning chats more than 2
I think `offset` and `limit` is not having any impact at all

Reply:
offset and limit are not ids of chat. they are more of filters. ex: if we have 1000 chats in the db and we want just the last 10, then, limit = 10; if we want 10 chats below 800, then, offset can be 200 and limit 10. They are provided by sequelize. and act just like filters

Reply:
so why Am I getting 5 chats, Even tho I provided `offset` of 102 and `limit` of 10.
there will be No validation for that????